code/game/atoms.dm

```diff
@@ -2079,11 +2079,19 @@
 
 	return TRUE
 
-//Update the screentip to reflect what we're hoverin over
 /atom/MouseEntered(location, control, params)
-	. = ..()
+	SSmouse_entered.hovers[usr.client] = src
+
+/// Fired whenever this atom is the most recent to be hovered over in the tick.
+/// Preferred over MouseEntered if you do not need information such as the position of the mouse.
+/// Especially because this is deferred over a tick, do not trust that `client` is not null.
+/atom/proc/on_mouse_enter(client/client)
```

Comment:
do you think all MouseEntered() overrides should be moved to executing via on_mouse_enter() or just the screentip stuff

Reply:
I think anything that doesn't need extra information like mouse position and stuff should, yeah. Just haven't made that movement here, especially since none of it is really expensive.